lib/plugins/aws/deploy/compile/events/apiGateway/lib/apiKeys.js

```diff
@@ -39,9 +39,9 @@ module.exports = {
         _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
           newApiKeyObject);
 
-        // Add API Key to Outputs section
+        // Add Id of API Key to Outputs section (Note: This is the Id, not the value!)
         const newOutput = {
```

Comment:
Yes. I thought that too. The problem is that we need to know which API keys belong to which Serverless service as the SDK will return all the API Keys in the users account (even from other services).

The way this fix is implemented is that it takes the IDs from the output, fetches all the API Keys from the users account and drops all the ones that are not relevant for the current service. Those are then added to the `apiKeys` array in the `info` object.

That's why this still needs to be around...

Maybe there's another way but I have not found another solution yet...